### ISSUE TITLE
Adjustable debounce timing

### ISSUE THREAD

**Issue by fusionstream** [bitfocus/companion#509]
An adjustable debounce timing in companion.

According this closed request: https://github.com/bitfocus/companion/issues/222, the issue with the streamdeck itself not having debounce was fixed with a firmware update. I experienced it once a few weeks back with a double trigger despite a firm press on the button. As I have 3 streamdecks and 2 were bought after the date in that issue, and were never connected to the official app, I don't know if my firmware is updated.

Nevertheless, I am still opening a new request because I sometimes also physically double press the button. It happens very rarely (twice during show; once, the button was also fully up) but the ability to set a null time would be great.

Aside from the use case I outlined above, we (a small group of operators and I) would like the ability to pass a single streamdeck to the show caller to trigger complete cues. If the show caller is nervous or does not have the skill to give the button a firm press, he/she may accidentally double press the button.

**Comment by fusionstream** [comment]
Unless the firmware was updated silently and very quickly, I can confirm that all my streamdecks are on firmware 1.0.170133 which I assume is the latest. And the double press event I listed above (firm solid press on my end) did trigger a double press.

**Comment by fusionstream** [comment]
This appears to occur with my older Streamdeck with serial number starting with AL19H1A... but not with the others.

**Comment by jerriley** [comment]
This issue happens with my new Stream Deck on latest firmware.

**Comment by DouggleCode** [comment]
My Stream Deck is version 1.02.005 and the bounce happens . So I can't use scene transitions on my ATEM Mini Pro. A tap during a transition pauses the transition so you get this half merged picture. I worked with Elgato support and they demonstrated that assigning text to each button and using the buttons in Word does not produce a double tap. It is obvious that Companion is "seeing" a double tap so I'm assuming they might have a de-bounce delay in their text plugin. Lost story short, because of that test, they say it is working properly. Hopefully Companion can add at least a hard coded debounce delay.

**Comment by josephdadams** [comment]
@DouggleCode are you using the stream deck app and plugin? I ask because I’ve personally seen this behavior sometimes when using the plugin.

**Comment by DouggleCode** [comment]
That is correct @josephdadams I have Stream Deck app 4.8.1 (13027) and the Companion plugin installed from the More Actions area.

**Comment by DouggleCode** [comment]
I don't see a double press happening in the Companion Log though.. I just know I can replicate what is happening when I double tap the Physical "Auto" button on the ATEM. I suppose the the mean time I can use CUT and that shouldn't have time to have a double tap. Of course if something is causing a double tap, you would not see a scene change due to the CUT button going to from input 1 to input 2 and then right back to 1

**Comment by josephdadams** [comment]
What version of Companion @DouggleCode ?

**Comment by DouggleCode** [comment]
@josephdadams Companion 1.4.0

**Comment by josephdadams** [comment]
Jump over to Companion 2.0 and try it. You’ll need the latest plugin for the native app. https://s3.bitfocus.io/builds/plugin/io.bitfocus.companion-plugin.streamDeckPlugin

**Comment by DouggleCode** [comment]
@josephdadams I was looking for the link for 2.0 but wasn't sure were to find it I see the source code downloads. I would need the pc and mac versions

**Comment by DouggleCode** [comment]
Got it! Thank you. I'll give that a try and see what happens. New to GitHub but getting my training wheels on. :)
